prov/rxm/src/rxm_cq.c

```diff
@@ -375,16 +373,14 @@ ssize_t rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
 
 ssize_t rxm_cq_handle_rx_buf(struct rxm_rx_buf *rx_buf)
 {
-	switch (rx_buf->pkt.ctrl_hdr.type) {
-	case ofi_ctrl_data:
+	if (OFI_LIKELY(rx_buf->pkt.ctrl_hdr.type == ofi_ctrl_data))
 		return rxm_cq_handle_data(rx_buf);
-	case ofi_ctrl_large_data:
+	else if (OFI_LIKELY(rx_buf->pkt.ctrl_hdr.type == ofi_ctrl_large_data))
```

Comment:
Should we just change this to else and have an assert for `ofi_ctrl_large_data`?

Reply:
Sure. This patch updated